packages/open-api-gateway/test/project/open-api-gateway-ts-project/regenerate-clients.test.ts

```diff
@@ -0,0 +1,336 @@
+      defaultReleaseBranch: "mainline",
+      name: "@test/my-api",
+      clientLanguages: [ClientLanguage.TYPESCRIPT],
+    });
+    project.synth();
+
+    const { generatedCodeDir, outdir } = project;
+    const generatedFileStats = fs.statSync(
+      path.join(
+        outdir,
+        generatedCodeDir,
+        ClientLanguage.TYPESCRIPT,
+        "src",
+        "index.ts"
+      )
+    );
+
+    // create another project with the same outdir
+    const projectOther = new OpenApiGatewayTsProject({
+      defaultReleaseBranch: "mainline",
+      name: "@test/my-api",
+      clientLanguages: [ClientLanguage.TYPESCRIPT],
+      outdir,
+    });
+    projectOther.synth();
+
+    const generatedFileStatsOther = fs.statSync(
+      path.join(
+        outdir,
+        generatedCodeDir,
+        ClientLanguage.TYPESCRIPT,
+        "src",
+        "index.ts"
+      )
+    );
+
+    // current `index.ts` should have the same modified time as the original project
+    expect(generatedFileStats.mtime).toEqual(generatedFileStatsOther.mtime);
+  });
+
+  it("forceGenerateCodeAndDocs=true will regenerate clients/docs", () => {
+    const project = new OpenApiGatewayTsProject({
+      defaultReleaseBranch: "mainline",
+      name: "@test/my-api",
+      clientLanguages: [ClientLanguage.TYPESCRIPT],
+      documentationFormats: [DocumentationFormat.HTML2],
+    });
+    project.synth();
+
+    const { generatedCodeDir, outdir } = project;
+    const generatedFileStats = fs.statSync(
+      path.join(
+        outdir,
+        generatedCodeDir,
+        ClientLanguage.TYPESCRIPT,
+        "src",
+        "index.ts"
+      )
+    );
+    const generatedDocHTML2Stats = fs.statSync(
+      path.join(
+        outdir,
+        generatedCodeDir,
+        "documentation",
+        DocumentationFormat.HTML2,
+        "index.html"
+      )
+    );
+
+    const projectOther = new OpenApiGatewayTsProject({
+      defaultReleaseBranch: "mainline",
+      name: "@test/my-api",
+      clientLanguages: [ClientLanguage.TYPESCRIPT],
+      documentationFormats: [DocumentationFormat.HTML2],
+      outdir,
+      forceGenerateCodeAndDocs: true,
+    });
+    projectOther.synth();
+
+    const generatedFileStatsOther = fs.statSync(
+      path.join(
+        outdir,
+        generatedCodeDir,
+        ClientLanguage.TYPESCRIPT,
+        "src",
+        "index.ts"
+      )
+    );
+    const generatedDocHTML2StatsOther = fs.statSync(
+      path.join(
+        outdir,
+        generatedCodeDir,
+        "documentation",
+        DocumentationFormat.HTML2,
+        "index.html"
+      )
+    );
+
+    // current `index.ts` should have later modified timestamp as the original project's
+    expect(generatedFileStats.mtime.getTime()).toBeLessThan(
+      generatedFileStatsOther.mtime.getTime()
+    );
+    // current `index.html` in "html2" doc folder should be untouched
+    expect(generatedDocHTML2Stats.mtime).toEqual(
+      generatedDocHTML2StatsOther.mtime
+    );
```

Comment:
Should the docs be regenerated too with `forceGenerateCodeAndDocs: true`?

Reply:
the generator is called with `--minimal-update` switch, so actually the output is not changed, even though the generator runs. I've added a comment in the code describing this behavior